Framework/PythonInterface/plugins/algorithms/SetSampleFromLogs.py

```diff
@@ -164,12 +166,18 @@ def PyExec(self):
         environment = self._createEnvironment(runObject, instrEnum)
 
         # let SetSample generate errors if anything is wrong
-        SetSample(InputWorkspace=wksp,
-                  Material=material,
-                  Geometry=geometry,
-                  Environment=environment,
-                  ContainerGeometry=geometryContainer,
-                  ContainerMaterial=materialContainer)
+        try:
+            SetSample(InputWorkspace=wksp,
+                      Material=material,
+                      Geometry=geometry,
+                      Environment=environment,
+                      ContainerGeometry=geometryContainer,
+                      ContainerMaterial=materialContainer)
+        except Exception as inst:
```

Comment:
Is this a condition that we can check for rather than doing `try`/`except`?

Reply:
That's my intention as well. But I am not sure how we can check whether we are dealing with vanadium run which is without container. It will be nice if you can help providing some clue here. Thank you!

Reply:
What is tripping it up, the `ContainerGeometry` and `ContainerMaterial`? The trick around that is to create a `dict` and add "acceptable" values to it. Something like:
```python
kwargs={}
if # something about container is valid:
  kwargs[`ContainerGeometry`] = # some value
  kwargs[`ContainerMaterial`] = # some other value
SetSample(InputWorkspace=wksp, Material=material, Environment=environment, **kwargs)
```
It would be even better if you add unit test to check show the problem

Reply:
Test the run with vanadium again I found it is the `Environment` input parameter that is causing the problem. For vanadium rod run, the `environment` parameter passed to `SetSampleFromLogs` is empty.